src/Json.Schema.ToDotNet/PropertyInfoDictionary.cs

```diff
@@ -609,15 +623,7 @@ private void OnAdditionalTypeRequired(CodeGenHint hint, JsonSchema schema)
         /// </remarks>
         internal TypeSyntax GetConcreteListType(string propertyName)
```

Comment:
>GetConcreteListType [](start = 28, length = 19)

The methods `GetConcreteListType` and `GetConcreteDictionaryType` used regexes unnecessarily, and also performed the same function in two different ways. Replace regex with a simple `Substring` call, and DRY the methods out with a helper.